### PR TITLE
app: show thumbnails in grid when available

### PR DESCRIPTION
**Pull request by ScottSWu**
Changes:
* add a ThumbnailOverlay class that just draws the thumbnail image to a canvas
* add the `thumbnails_only` parameter (though hardcoded to false for now)
* expose `dataset.app_config` to various places in the code (needed to access `grid_media_thumbnails`)

FiftyOne thumbnail PRs:
* add option to generate thumbnails for FiftyOne
* update example script to generate thumbnails
* add custom grid_thumbnail_fields field to server backend (#9)
* add custom thumbnails_only param to /samples server route (#10)
* add custom sample_id param to server route (#11)
* **show thumbnails in the grid view of the app** (#12)
* lazy load the full sample when opened in a modal (#13)

https://app.asana.com/0/0/1203821999397249/f

### CODE REVIEW

File: app/packages/core/src/components/Grid/useExpandSample.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import { FlashlightConfig } from "@fiftyone/flashlight";
 import { useCallback } from "react";
 
+import { FlashlightConfig } from "@fiftyone/flashlight";
```
Comment (konstantin-azarov) [inline app/packages/core/src/components/Grid/useExpandSample.ts]: Why this change?

**Comment by ScottSWu** [comment]
Will rebase this one

File: app/packages/looker/src/overlays/thumbnail.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    const [tlx, tly] = t(state, 0, 0);
+    const [brx, bry] = t(state, 1, 1);
+    const tmp = ctx.globalAlpha;
+    ctx.globalAlpha = state.options.alpha;
+    ctx.drawImage(this.image, tlx, tly, brx - tlx, bry - tly);
+    ctx.globalAlpha = tmp;
+  }
+
+  // There is no need to support the remaining methods, as the ThumbnailOverlay
+  // won't be used in a modal, and won't need to handle any interactions.
```
Comment (konstantin-azarov) [inline app/packages/looker/src/overlays/thumbnail.ts:55]: Should these methods throw NotImplemented instead?

File: app/packages/state/src/hooks/useCreateLooker.ts [hunk 3]
```diff
@@ -30,7 +34,8 @@ export default <T extends FrameLooker | ImageLooker | VideoLooker>(
   const activeId = isModal ? useRecoilValue(mainGroupSample)._id : null;
 
   const view = useRecoilValue(viewAtoms.view);
-  const dataset = useRecoilValue(datasetName);
+  const dataset = useRecoilValue(datasetAtom);
+  const datasetName = useRecoilValue(datasetNameAtom);
```
Comment (konstantin-azarov) [inline app/packages/state/src/hooks/useCreateLooker.ts]: What's the point of this whole `datasetNameAtom` change?

**Review by konstantin-azarov — APPROVED** [review]

File: app/packages/core/src/components/Grid/useExpandSample.ts (hunk 1 above)
Comment (ScottSWu) [reply app/packages/core/src/components/Grid/useExpandSample.ts]: Accidentally left in when splitting up the PRs. I'll move it to the next PR.

File: app/packages/looker/src/overlays/thumbnail.ts (hunk 2 above)
Comment (ScottSWu) [reply app/packages/looker/src/overlays/thumbnail.ts:55]: They'll still get called, so they should return empty values. Updated the comment.

File: app/packages/state/src/hooks/useCreateLooker.ts (hunk 3 above)
Comment (ScottSWu) [reply app/packages/state/src/hooks/useCreateLooker.ts]: For some reason I thought `dataset` didn't include the name? Not sure, reverted.
